src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -789,30 +789,212 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
+                                                'DarkRed' { ""`e[2;31m"" }
+                                                'DarkGreen' { ""`e[2;32m"" }
+                                                'DarkYellow' { ""`e[2;33m"" }
+                                                'DarkBlue' { ""`e[2;34m"" }
+                                                'DarkMagenta' { ""`e[2;35m"" }
+                                                'DarkCyan' { ""`e[2;36m"" }
+                                                'Gray' { ""`e[2;37m"" }
+                                                'DarkGray' { ""`e[1;30m"" }
+                                                'Red' { ""`e[1;31m"" }
+                                                'Green' { ""`e[1;32m"" }
+                                                'Yellow' { ""`e[1;33m"" }
+                                                'Blue' { ""`e[1;34m"" }
+                                                'Magenta' { ""`e[1;35m"" }
+                                                'Cyan' { ""`e[1;36m"" }
+                                                'White' { ""`e[1;37m"" }
+                                                default { ""`e[1;31m"" }
+                                            }
+                                        }
+
+                                        # return length of string sans VT100 codes
+                                        function Get-RawStringLength($string) {
+                                            $vtCodes = ""`e[0m"", ""`e[2;30m"", ""`e[2;31m"", ""`e[2;32m"", ""`e[2;33m"", ""`e[2;34m"",
+                                                ""`e[2;35m"", ""`e[2;36m"", ""`e[2;37m"", ""`e[1;30m"", ""`e[1;31m"", ""`e[1;32m"",
+                                                ""`e[1;33m"", ""`e[1;34m"", ""`e[1;35m"", ""`e[1;36m"", ""`e[1;37m""
+
+                                            $newString = $string
+                                            foreach ($vtCode in $vtCodes) {
+                                                $newString = $newString.Replace($vtCode, '')
+                                            }
+
+                                            return $newString.Length
+                                        }
+
+                                        # returns a string cut to last whitespace
+                                        function Get-TruncatedString($string, [int]$length) {
+                                            if ($string.Length -le $length) {
+                                                return $string
+                                            }
+
+                                            if (-not $string.Contains(' ')) {
+                                                return $string.Substring(0, $length)
+                                            }
+
+                                            $split = $string.Substring(0, $length).Split(' ')
```

Comment:
there are a couple of bugs here - you're going to be including spaces at the end of the line (you probably shouldn't use `splitoptions` because multiple spaces could be anywhere. Also if the string includes a tab, you'll be producing a too long line. Here's a possible solution:
```powershell
param ( $s, $l )
$tabString = $s.Replace("`t","    ")
if ( $tabString.length -le $l ) { return $tabString }
$ss = $tabString.substring(0,$l)
$r = [regex]"\s*$"
$m = $r.Match($ss)
$ns = $ss.Substring(0,$m.Index)
return $ns
```

Reply:
We discussed and using `-split '\s', -2` instead